File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        def check(st):
            i=0
            j=len(st)-1
            while i<j:
                if st[i]!=st[j]:
                    return False
                i+=1
                j-=1
            return True
        if check(s):
            return s
        i=0
        j=len(s)-1
        res=""
        while i<len(s):
            for j in range(len(s),i,-1):
                st=s[i:j]
                if check(st):
                    if len(st)>len(res):
                        res=st
            i+=1
        return res

        

        

File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    pair<int,int> check(string& s,int l,int r){
        while(l>=0 && r<s.size() && s[l]==s[r]){
            l--;
            r++;
        }
        return {r-l-1,l+1};
    }
    string longestPalindrome(string s) {
        int ans=0,j=-1;
        for(int i=0;i<s.size();i++){
            auto a=check(s,i,i);
            auto b=check(s,i,i+1);
            if(a.first>ans){
                ans=a.first;
                j=a.second;
            }
            if(b.first>ans){
                ans=b.first;
                j=b.second;
            }
        }
        return s.substr(j,ans);
    }
};